floss/stackstrings.py

```diff
@@ -57,10 +64,33 @@ def apicall(self, emu, op, pc, api, argv):
         stack_buf = emu.readMemory(stack_top, stack_size)
         self.ctxs.append(CallContext(op.va, stack_top, stack_bottom, stack_buf))
 
-
-def extract_call_contexts(vw, fva):
+    def posthook(self, emu, op, endpc):
+        self.get_context_via_mov_heuristic(emu, op, endpc)
+
+    def get_context_via_mov_heuristic(self, emu, op, endpc):
+        """ Extract contexts at end of a basic block (bb) if bb contains enough movs to stack memory. """
+        # TODO check number of written bytes?
+        # count movs, shortcut if this basic block has enough writes to trigger context extraction already
+        if self.mov_count < MIN_NUMBER_OF_MOVS and self.is_stack_mov(emu, op):
+            self.mov_count += 1
+        if endpc in self.bb_ends:
+            if self.mov_count >= MIN_NUMBER_OF_MOVS:
+                self.extract_context(emu, op)
+            # reset counter at end of basic block
+            self.mov_count = 0
+
+    def is_stack_mov(self, emu, op):
```

Comment:
this function doesn't actually check to see if the destination is on the stack. do you think we should do this? i guess its pruned off elsewhere (in the ctx collection).